pkg/controllers/termination/controller.go

```diff
@@ -77,11 +77,15 @@ func (c *Controller) Finalize(ctx context.Context, node *v1.Node) (reconcile.Res
 		return reconcile.Result{}, fmt.Errorf("cordoning node, %w", err)
 	}
 	if err := c.terminator.Drain(ctx, node); err != nil {
-		if terminator.IsNodeDrainError(err) {
-			c.recorder.Publish(terminatorevents.NodeFailedToDrain(node, err))
+		if !terminator.IsNodeDrainError(err) {
```

Comment:
Is there a reason to make an extra `Delete` call if we already know from the `Get` that the machine doesn't exist?

Reply:
It should be idempotent and batched. I can add in logic that just patches out the finalizer if you think it's necessary.

Reply:
I was mainly just thinking that the extra API call wasn't necessary